foundations/javascript_basics/object_basics.md

```diff
@@ -20,6 +20,94 @@ Objects are a *very* important part of the JavaScript language, and while for th
+
+Besides the formal differences, there are also some technical differences, which affect how we use each data type.
+When you define a primitive variable, it will contain a copy of the information provided to it:
+
+```javascript
+let data = 42;
+// dataCopy will store a copy of what data contains, so a copy of 42
+let dataCopy = data;
+
+// which means that making changes to dataCopy won't affect data
+dataCopy = 43;
+
+console.log(data); // 42
+console.log(dataCopy); // 43
+```
+
+On the other hand, when you define an object variable, it will contain a *reference* to the object provided to it:
+
+```javascript
+// obj contains a reference to the object we defined on the right side
+const obj = { data: 42 };
+// objCopy will contain a reference to the object referenced by obj
+const objCopy = obj;
+
+// making changes to objCopy will make changes to the object that it refers to
+objCopy.data = 43;
+
+console.log(obj); // { data: 43 }
+console.log(objCopy); // { data: 43 }
+```
+
+This behavior isn't new to you. In your last project, you made changes to the cells in the Etch-A-Sketch grid by using references.
+Let's take this code snippet as an example:
+
+```javascript
+const element = document.querySelector("#container");
+element.style.backgroundColor = "red";
+```
+
+We're mutating the variable we declared (`element`), yet the changes affect the corresponding node in the DOM.
+Why does it happen?
+That's because the node we have in our code is a **reference** to the same node that our DOM uses. If that wasn't a reference, but a copy like primitive data types behave, our changes would have **no** effect! Because the changes would be made to the local copy we have.
+
+This behavior is also something to consider when we pass arguments to a function, let's take the following functions for example:
+
+```javascript
+function increaseCounterObject(objectCounter) {
+  objectCounter.counter += 1;
+}
+
+function increaseCounterPrimitive(primitiveCounter) {
+  primitiveCounter += 1;
+}
+
+const object = { counter: 0 };
+let primitive = 0;
+
+increaseCounterObject(object);
+increaseCounterPrimitive(primitive);
+```
+
+Take a moment and guess what will happen to `object` and what will happen to `primitive` after we make the function calls.
+
+If you answered that the object counter would increase by 1, and the primitive counter wouldn't change, you're correct.
+Remember that `objectCounter` contains a reference to the same object as the `object` variable we gave it, while `primitiveCounter` contains only a copy of the primitive value only.
+
+<div class="lesson-note" markdown="1">
+
+#### Reassigning object data type variables
+
+While changing the object we have a reference to will change that object for all other variables that reference it, changing what we refer to won’t affect what the other variables refer to. This example should clarify it:
```

Comment:
Again, preferring the explicit mutate/reassign terminology than using "changing" twice with two different meanings in the same sentence.
```suggestion
While mutating the object we have a reference to will affect all other variables that reference it, reassigning what a variable refers to won’t affect what the other variables refer to. For example:
```